common/util.go

```diff
@@ -80,6 +80,22 @@ func RemoveContents(directory string) error {
 	return nil
 }
 
+// isDirEmpty checks if given directory is empty
+func isDirEmpty(name string) (bool, error) {
+	f, err := os.Open(name)
+	if err != nil {
+		return false, err
+	}
+	defer f.Close()
```

Comment:
That's pretty cool.

Reply:
Definitely useful, though for arguably one of the most potential useful usecases (`defer mux.Unlock()`) the performance penalty is too significant to be of use